src/Tooltip.js

```diff
@@ -79,10 +89,24 @@ class Tooltip extends React.Component {
     };
   }
 
+  show() {
+    if (!this.props.isOpen) {
+      this.toggle();
+    }
+  }
+  hide() {
+    if (this.props.isOpen) {
+      this.toggle();
+    }
+  }
+
   handleDocumentClick(e) {
     if (e.target === this._target || this._target.contains(e.target)) {
-      if (this._hoverTimeout) {
-        clearTimeout(this._hoverTimeout);
+      if (this._hideTimeout) {
+        clearTimeout(this._hideTimeout);
+      }
+      if (this._showTimeout) {
+        clearTimeout(this._showTimeout);
```

Comment:
You need to clear out `this.showTimeout` and `this.hideTimeout`. You can simply set them to `undefined`. Without clearing them out, they will still be truthy and thus, every click after the have been set will trigger the `clearTimeout`, even if the timeout has already been cleared.